packages/expo-updates/ios/EXUpdates/Multipart/EXUpdatesMultipartStreamReader.m

```diff
@@ -0,0 +1,136 @@
+//  Copyright © 2019 650 Industries. All rights reserved.
+
+#import <EXUpdates/EXUpdatesMultipartStreamReader.h>
+
+#define CRLF @"\r\n"
+
+@implementation EXUpdatesMultipartStreamReader {
+  __strong NSInputStream *_stream;
+  __strong NSString *_boundary;
+}
+
+- (instancetype)initWithInputStream:(NSInputStream *)stream boundary:(NSString *)boundary
+{
+  if (self = [super init]) {
+    _stream = stream;
+    _boundary = boundary;
+  }
+  return self;
+}
+
+- (NSDictionary<NSString *, NSString *> *)parseHeaders:(NSData *)data
+{
+  NSMutableDictionary *headers = [NSMutableDictionary new];
+  NSString *text = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+  NSArray<NSString *> *lines = [text componentsSeparatedByString:CRLF];
+  for (NSString *line in lines) {
+    NSUInteger location = [line rangeOfString:@":"].location;
+    if (location == NSNotFound) {
+      continue;
+    }
+    NSString *key = [line substringToIndex:location];
+    NSString *value = [[line substringFromIndex:location + 1] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+    [headers setValue:value forKey:key];
+  }
+  return headers;
+}
+
+- (void)emitChunk:(NSData *)data headers:(NSDictionary *)headers callback:(RCTMultipartCallback)callback done:(BOOL)done
+{
+  NSData *marker = [CRLF CRLF dataUsingEncoding:NSUTF8StringEncoding];
```

Comment:
i see what this does but lol

Reply:
yeah this one confused me for a bit as well. hesitant to change though since this class is a fork.